forge-gui/res/cardsfolder/upcoming/lamplight_phoenix.txt

```diff
@@ -0,0 +1,11 @@
+Name:Lamplight Phoenix
+ManaCost:1 R R
+Types:Creature Phoenix
+PT:3/3
+K:Flying
+T:Mode$ ChangesZone | Origin$ Battlefield | Destination$ Graveyard | ValidCard$ Card.Self | Execute$ TrigExile | TriggerDescription$ When CARDNAME dies, you may exile it and collect evidence 4. If you do, return Lamplight Phoenix to the battlefield tapped. (To collect evidence 4, exile cards with total mana value 4 or greater from your graveyard.)
+SVar:TrigExile:DB$ ChangeZone | Defined$ TriggeredNewCardLKICopy | Origin$ Graveyard | Destination$ Exile | UnlessCost$ CollectEvidence<4> | RememberChanged$ True | SubAbility$ DBReturn
```

Comment:
I think you should remove this line and make the Return line AB$ ChangeZone + Cost$ ExileFromGrave<1/CARDNAME> CollectEvidence<4>

Reply:
Because the Exile is part of the cost of the return

Reply:
good idea

Reply:
Hmm this way it allows you to exile itself with CellectEvidence. I don't think that's allowed ?

Reply:
We can change the priority of ColelctEvidence to happen after ExileFromGrave

Reply:
Currently it doesn't return

Reply:
Hmm not really. Doesn't even allow me to pay the cost

Reply:
check e.g. _Arena Rector_ for the right way to write this cost (the logic is currently a bit roundabout because of the zone change trigger origin)

Reply:
Hmn, not sure I follow

Reply:
Oh sorry this should be comment on the other PR. 

Anyhow now it's working

Reply:
using the paidList for the Defined should be cleaner (ExiledCards)

Reply:
Thanks, working